src/user/mod.rs

```diff
@@ -128,8 +127,7 @@ impl<'a> NewUser<'a> {
                     .optional()?;
 
                 if let Some(token) = token {
-                    send_user_confirm_email(user_email, &user.gh_login, &token)
-                        .map_err(|_| NotFound)?;
+                    send_user_confirm_email(user_email, &user.gh_login, &token);
```

Comment:
If there isn't an email, we shouldn't even be in this block because of the `if let Some(user_email) = user.email.as_ref() {` above, right? And if the email is present but not valid, we don't find that out synchronously. [Emails get put in mailgun's queue](https://documentation.mailgun.com/en/latest/user_manual.html#message-queue) and then their dashboard shows email sending success/failure.

Reply:
So what I think what we were seeing here is that the email address is being rejected on our end (probably in `lettre`) before we even generate an email and give it to mailgun.  For instance, if somebody manages to put `invalid` as their email address in GitHub.

Reply:
As a more specific example, here is the email address that was giving one user issues: `String.Format(\"{0}.{1}@live.com\", FirstName, LastName)`.  After looking at the profile a dozen times, I only just now realized that this is the email address on their profile page.

Reply:
That example would make a great test case and would certainly clarify this PR that it's not *sending* the email that fails, but that there's verification of an email address's validity in lettre that fails and returns an error. That wasn't at all clear from the original PR.